### ISSUE TITLE
Dialog Component Accessibility Issue

### ISSUE THREAD

**Issue by aubreyquinn** [microsoft/fluentui#18599]
I am using this framework: "@fluentui/react-northstar": "^0.53.0"

I am using the Dialog component. [Fluent UI - Dialog (windows.net)](https://fluentsite.z22.web.core.windows.net/0.53.0/components/dialog/definition)

When I run storybook, it fails the accessibility tests due to the Dialog triggering button.

Even when I use the "trapFocus" prop it does not pass.

My trigger button is causing the accessibility check to fail.
 
![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/80953505/122421048-04e9dc80-cf84-11eb-9595-643872454e14.png)

code snippet:

```
const dialogProps: DialogProps = {
    className:cancelEventForm,
    defaultOpen:props.shouldDefaultOpen,
    trapFocus:true,
    header:<Textcontent={`Cancel ${eventTypeDesc}?`}/>,
    content: (
        <Flex columngap="gap.medium">
            <Textsize="small"content={`This will cancel your ${eventTypeDesc} and attendees will be notified.`}/>
            <TextArea
                placeholder="If you want, say why you're canceling here"
                onChange={(event, data) =>setMessage(data?.value ?? "")}
                />
        </Flex>
    ),
    cancelButton:"Never mind",
    confirmButton:"Yes, cancel",
    onConfirm: () =>props.onSubmit(props.eventId, sanitizeHtml(message)),
    trigger:props.triggerButton
};
 
return<Dialog{...dialogProps}/>;
```
 

> I was told that the only way to fix this is kind of a hack.
> 
> If the focus is trapped inside that dialog then this might be intentional. Generally aria-modal should be used with modal dialogs, however it does not have great cross browser support. Therefore the 'solution/hack' is to apply aria-hidden on all other page elements when the dialog is open so that screen readers cannot find them in virtual cursor mode.
> 
> the component powering the focus trap in the dialog is called FocusTrapZonewhich will set aria-hidden quite indiscriminately on the body. However until aria-modal has full support, the aria-hidden hack will prolly be around to stay.

**Comment by ling1726** [comment]
@kolaps33  AFAIK `FocusTrapZone` will apply `aria-hidden` on all non-trapped content. This is expected since, other content shouldn't be visible to the virtual cursor. 

We could use `aria-modal` for this, but it's not well supported https://caniuse.com/?search=aria-modal

**Comment by kolaps33** [comment]
I am already in the call with Aubrey and figure out that aria-hidden=true is hiding the trigger button.
But this is apply only when dialog is open. This is expected implementation of dialog coming from Fluent UI.
Explanation: we set aria-hidden=true on other elements outside the dialog, to disable Virtual Cursor navigation outside the dialog...
We verified that aria-hidden=true is removed once dialog is closed. 

Closing as by design.
